src/components/CurrentUser/CurrentUser.jsx

```diff
@@ -1,12 +1,102 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import { getUser } from '../../api';
 import './CurrentUser.scss';
 
-export const CurrentUser = () => (
-  <div className="CurrentUser">
-    <h2 className="CurrentUser__title"><span>Selected user: 2</span></h2>
+export class CurrentUser extends React.Component {
+  state = {
+    user: null,
+    isLoading: false,
+    userError: false,
+    hasLoadingError: false,
+  }
 
-    <h3 className="CurrentUser__name">Ervin Howell</h3>
-    <p className="CurrentUser__email">Shanna@melissa.tv</p>
-    <p className="CurrentUser__phone">010-692-6593 x09125</p>
-  </div>
-);
+  componentDidMount() {
+    this.loadData();
+  }
+
+  componentDidUpdate(prevProps) {
+    const { userId } = this.props;
+
+    if (prevProps.userId !== userId) {
+      this.loadData();
+    }
+  }
+
+  async loadData() {
+    const { userId } = this.props;
+
+    this.setState({
+      isLoading: true,
+      hasLoadingError: false,
+    });
+
+    try {
+      const user = await getUser(userId);
+
+      if (!user) {
+        this.setState({ userError: true });
+      }
+
+      this.setState({
+        user,
+        isLoading: false,
+      });
+    } catch (error) {
+      this.setState({
+        hasLoadingError: true,
+      });
+    }
+  }
+
+  render() {
+    const {
+      user,
+      isLoading,
+      userError,
+      hasLoadingError,
+    } = this.state;
+    const { clearUser } = this.props;
+
+    return (
+      <div className="CurrentUser">
+        {
+          (
+            isLoading && <h3>Loadings</h3>,
+            hasLoadingError && <h3>Error occurred during loading</h3>,
+            userError && <h3>Can&apos;t find user</h3>
+          )
+        }
```

Comment:
this works not as you expected, it always shows 'Loading', even if there were LoadingError or User Error

![image](https://user-images.githubusercontent.com/57598825/98437586-0436c100-20ec-11eb-8808-6110479ed336.png)